src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php

```diff
@@ -402,6 +403,15 @@ public function get_item( $request ) {
 			return $revision;
 		}
 
+		if ( $parent->ID !== $revision->post_parent ) {
+			return new WP_Error(
+				'rest_post_invalid_parent',
+				/* translators: %s: A post revision id. */
+				sprintf( __( 'Parent post does not have a revision with id of "%s"' ), $revision->post_parent ),
```

Comment:
The error message mentions the id as being the revision id, but the passed in value is the revision's `post_parent`. Should `$revision->ID` (or `$request['id']`) be passed in instead, or adjust the error message to a more generic `'Parent post does not have the requested revision id'`?

Reply:
That was an oversight, thanks @andrewserong! I'll update.

I was trying to decide whether to highlight in invalid parent (with respect to the supplied revision id) or invalid revision id (with respects to its parent), and my brain exploded.

Reply:
Thanks for updating!